### PR TITLE
4 as a user i need to be able to sign upsign in to the plateform using my google account

### PR DESCRIPTION
**Pull request by youcef-s**
The user can sign in/up using his Google account, and stay logged in as long as the access token has not expired, when the access token is expired he/she can request a new one by sending an Get request to '/refresh' if the refresh token provided is correct.

### CODE REVIEW

File: backend/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	});
+  }
+
+  @Get('profile')
+  @UseGuards(JwtGuard)
+  profile(@Req() req: Request) {
+	return req.user;
+  }
+
+  @Get('refresh')
```
Comment (youssefachehboune) [inline backend/src/auth/auth.controller.ts:39]: /refresh does not update the jwt cookie .

**Review by youssefachehboune — CHANGES_REQUESTED** [review]

File: backend/src/auth/auth.controller.ts (hunk 1 above)
Comment (youcef-s) [reply backend/src/auth/auth.controller.ts:39]: solved!

File: backend/src/auth/strategies/google.strategy.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		data: {
+		  email: googleuser.email,
+		  twoFactorEnabled: false,
+		  role: 'USER',
+		},
+	  });
+	  await prisma.userProfile.create({
+		  data: {
+			  user_id: dbuser.id,
+			  username: googleuser.email.substring(0, googleuser.email.indexOf('@')),
```
Comment (youssefachehboune) [inline backend/src/auth/strategies/google.strategy.ts:41]: if the username for a Google account user  is already being used as a  Intra login  Prisma gets complaining

**Review by youssefachehboune — CHANGES_REQUESTED** [review]

File: backend/src/auth/auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+@Controller()
+export class AuthController {
+  constructor(private AuthService: AuthService) {}
+
+  @Get('google')
+  @UseGuards(GoogleGuard)
+  async googleAuth() {}
+
+  @Get('google/redirect')
+  @UseGuards(GoogleGuard)
```
Comment (youssefachehboune) [inline backend/src/auth/auth.controller.ts:17]: When I log in and get redirected from the server, I attempt to refresh the same page, but I encounter an exception from Google Guard.

**Review by youssefachehboune — CHANGES_REQUESTED** [review]

**Review by youssefachehboune — APPROVED** [review]

**Pull request merged**